### ISSUE TITLE
adds spurious data to sessionstore.js file, can interfere with restore

### ISSUE THREAD

**Issue by sumyungguy** [senicar/enhanced-middle-click#5]
With enhanced middle click installed, extra "tabview-tabs" and "tabview-groups" data is added to the sessionstore.js file. In some cases this can cause session restore to malfunction.

Here is a simple sessionstore.js, created in Firefox 42 with one window open showing the Firefox start page. It was made with a new clean profile, all default preferences except "show my windows and tabs from last time".

 With no addons installed:

{"windows":[{"tabs":[{"entries":[{"url":"about:home","title":"Mozilla Firefox Start Page","ID":2,"docshellID":5,"docIdentifier":2,"persist":true}],"lastAccessed":1450844431666,"hidden":false,"attributes":{},"index":1,"image":"chrome://branding/content/icon32.png"}],"selected":1,"_closedTabs":[],"busy":false,"width":1413,"height":792,"screenX":398,"screenY":67,"sizemode":"normal"}],"selectedWindow":1,"_closedWindows":[],"session":{"lastUpdate":1450844431757,"startTime":1450844423670,"recentCrashes":0},"global":{}}

With enhanced middle click installed:

{"windows":[{"tabs":[{"entries":[{"url":"about:home","title":"Mozilla Firefox Start Page","ID":2,"docshellID":5,"docIdentifier":2,"persist":true}],"lastAccessed":1450844323565,"hidden":false,"attributes":{},"extData":{"tabview-tab":"{\"groupID\":2,\"active\":true}"},"index":1,"image":"chrome://branding/content/icon32.png"}],"selected":1,"_closedTabs":[],"busy":false,"extData":{"tabview-groups":"{\"nextID\":3,\"activeGroupId\":2,\"totalNumber\":1}","tabview-group":"{\"2\":{\"bounds\":{\"left\":15,\"top\":5,\"width\":908.4540000000001,\"height\":508.92100000000005},\"userSize\":null,\"title\":\"\",\"id\":2}}","tabview-ui":"{\"pageBounds\":{\"left\":0,\"top\":0,\"width\":1413,\"height\":783}}"},"width":1413,"height":792,"screenX":398,"screenY":67,"sizemode":"normal"}],"selectedWindow":1,"_closedWindows":[],"session":{"lastUpdate":1450844323642,"startTime":1450844211311,"recentCrashes":0},"global":{}}

In some (rare) cases, this interferes with session restore. In Firefox 43, certain windows that were already closed, will be incorrectly re-opened after restart if enhanced middle click is installed, but not when it's not. This is most likely caused by an interaction with a new bug in Firefox 43 (which I'm trying to pin down), and not necessarily a bug in enhanced middle click. However, I think that the sessionstore.js data should be the same, whether enhanced middle click is installed or not.

**Comment by sumyungguy** [comment]
Just saw that Tab Groups will be removed in Firefox 45, which will also be the new ESR release. I guess there's no real point in trying to fix this then... nevermind.

**Comment by senicar** [comment]
@sumyungguy I looked into the issue and why it happens and I found that it was caused by a fix that solved another bug some time ago, when Tab Groups functionality had to be initiated manually if I wanted to get access to Tab Groups and tab groups names.

However, as you pointed it out, tab groups will be removed in the future version, so I will sadly have to remove this functionality from the addon as well.

I would still like to thank you for reporting the issue and I apologize for not responding sooner.

Regards
